datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/util/BirdcageOptimizer.scala

```diff
@@ -132,7 +133,9 @@ object BirdcageOptimizer extends RuleExecutor[LogicalPlan] {
     Batch(
       "RewriteSubquery", Once,
       RewritePredicateSubquery,
-      CollapseProject) :: Nil
+      CollapseProject) ::
+    Batch(
+      "MVProjectAdd", Once, MVProjectColumnsAdd)::Nil
```

Comment:
I feel it is like a  workaround, I think the fix should come from SQL generation layer of MV to support this combination of nodes.

Reply:
current logic:
MVqueryString->logicalPlan->modularize->optimizer->modularPlan

modularPlan.isSPJGH -> false(not support groupBy node without child of select node) -> throw unsupport exception

the key is that when generating logicalPlan, Spark use rule `RemoveRedundantProject` to remove useless project when optimizing. What Qiu is doing is to revert that rule in MV  optimizer

Reply:
@kevinjmh It is not that he is removing a useless project, he is adding a project so that sql generation in mv core will not fail. Instead of solving the failure in SQL generation, he is trying to add another project as a workaround.
What I am asking is better fix the issue than adding the workarounds.

Reply:
if so, `isSPJGH` should not block this mv query.  What do you think of https://github.com/apache/carbondata/pull/3072#issuecomment-485332427

Reply:
Yes `isSPJGH` can allow the groupy+relation combination.

Reply:
after #3072 (comment) done and then add one  case g@modular.GroupBy(_, _, _, _, s@modular.Select(_, _, _, _, _, _, _, _, _, _), _, _, _) to the function extractSimpleOperator, it works.

Reply:
@ravipesala  modify as zhxiaoping says,  please review again